dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -4203,35 +4204,48 @@ else if( field.getFieldType().equals(Field.FieldType.CATEGORY.toString()) ) {
                     }
                 }
             }
-            if(field.isUnique()){
-                try{
+            if (field.isUnique()) {
+                try {
                     StringBuilder buffy = new StringBuilder();
 
                     buffy.append(" +(live:true working:true)");
                     buffy.append(" +structureInode:" + contentlet.getStructureInode());
-                    buffy.append(" +languageId:" + contentlet.getLanguageId());
                     buffy.append(" +(working:true live:true)");
-                    if(UtilMethods.isSet(contentlet.getIdentifier())){
+                    if (UtilMethods.isSet(contentlet.getIdentifier())) {
                         buffy.append(" -(identifier:" + contentlet.getIdentifier() + ")");
+                        buffy.append(" +languageId:" + contentlet.getLanguageId());
                     }
-                    buffy.append(" +" + contentlet.getStructure().getVelocityVarName() + "." + field.getVelocityVarName() + ":\"" + escape(getFieldValue(contentlet, field).toString()) + "\"");
+                    buffy.append(" +" + contentlet.getStructure().getVelocityVarName() + "." + field
+                            .getVelocityVarName() + ":");
+                    buffy.append(
+                            (field.getDataType().contains(DataTypes.INTEGER.toString()) || field
+                                    .getDataType().contains(DataTypes.FLOAT.toString())) ? escape(
+                                    getFieldValue(contentlet, field).toString())
+                                    : "\"" + escape(getFieldValue(contentlet, field).toString())
+                                            + "\"");
                     List<ContentletSearch> contentlets = new ArrayList<ContentletSearch>();
                     try {
-                        contentlets = searchIndex(buffy.toString(), -1, 0, "inode", APILocator.getUserAPI().getSystemUser(), false);
+                        contentlets = searchIndex(buffy.toString(), -1, 0, "inode",
+                                APILocator.getUserAPI().getSystemUser(), false);
                     } catch (Exception e) {
-                        Logger.error(this, e.getMessage(),e);
-                        throw new DotContentletValidationException(e.getMessage(),e);
+                        Logger.error(this, e.getMessage(), e);
+                        throw new DotContentletValidationException(e.getMessage(), e);
                     }
                     int size = contentlets.size();
-                    if(size > 0 && !hasError){
+                    if (size > 0 && !hasError) {
 
                         Boolean unique = true;
                         for (ContentletSearch contentletSearch : contentlets) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)